File: 프로그래머스/lv2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.cpp
#include <string>
#include <vector>

using namespace std;

string solution(string s) {
    for(int i=0; i< s.length(); i++)
    {
        int first = 0;
        while(i < s.length() && s[i] != ' ')
        {
            if (first == 0 && s[i] >= 'a' && s[i] <= 'z')
            {
                first = 1;
                s[i] -= 32;
            }   
            else if (first == 1 && s[i] >= 'A' && s[i] <= 'Z')
            {
                s[i] += 32;
            }
            first = 1;
            i++;
        }
    }
    return s;
}

File: 프로그래머스/2/12951. JadenCase 문자열 만들기/JadenCase 문자열 만들기.java
class Solution {
    public String solution(String s) {
        String[] answer = s.toLowerCase().split(" ", -1);
        for (int i = 0; i < answer.length; i++) {
            if (answer[i].length() > 0) {
                answer[i] = answer[i].substring(0, 1).toUpperCase() + answer[i].substring(1);
            }
        }
        return String.join(" ", answer);
    }
}
